wasm/validator/src/visitor.rs

```diff
@@ -0,0 +1,747 @@
+        validate_transfer_asset(&Transfer<Asset, NumericValue, Asset>),
+
+        // Validate SetKeyValueBox
+        validate_set_domain_key_value(&SetKeyValue<Domain, Name, Value>),
+        validate_set_account_key_value(&SetKeyValue<Account, Name, Value>),
+        validate_set_asset_definition_key_value(&SetKeyValue<AssetDefinition, Name, Value>),
+        validate_set_asset_key_value(&SetKeyValue<Asset, Name, Value>),
+
+        // Validate RemoveKeyValueBox
+        validate_remove_domain_key_value(&RemoveKeyValue<Domain, Name>),
+        validate_remove_account_key_value(&RemoveKeyValue<Account, Name>),
+        validate_remove_asset_definition_key_value(&RemoveKeyValue<AssetDefinition, Name>),
+        validate_remove_asset_key_value(&RemoveKeyValue<Asset, Name>),
+
+        // Validate GrantBox
+        validate_grant_account_permission(&Grant<Account, PermissionToken>),
+        validate_grant_account_role(&Grant<Account, RoleId>),
+
+        // Validate RevokeBox
+        validate_revoke_account_permission(&Revoke<Account, PermissionToken>),
+        validate_revoke_account_role(&Revoke<Account, RoleId>),
+
+        // Validate UpgradeBox
+        validate_upgrade_validator(&Upgrade<Validator>),
+    }
+}
+
+/// Default validation for [`SignedTransaction`].
+///
+/// Performs execution if transaction contains [`Executable::Instructions`]
+/// and does nothing if [`Executable::Wasm`].
+///
+/// Execution is done to properly validate dependent instructions.
+#[allow(clippy::needless_pass_by_value)]
+pub fn validate_transaction<V: Validate + ?Sized>(
+    validator: &mut V,
+    authority: &AccountId,
+    transaction: SignedTransaction,
+) -> Verdict {
+    let verdict = validator.validate_transaction_signatures(authority, transaction.signatures());
+
+    if verdict.is_deny() {
+        return verdict;
+    }
+
+    validator.validate_transaction_payload(authority, transaction.payload())
+}
+
+pub fn validate_transaction_payload<V: Validate + ?Sized>(
+    validator: &mut V,
+    authority: &AccountId,
+    payload: &TransactionPayload,
+) -> Verdict {
+    match payload.instructions() {
+        Executable::Wasm(wasm) => validator.validate_wasm(authority, wasm),
+        Executable::Instructions(instructions) => {
+            for isi in instructions {
+                let verdict = validator.validate_instruction(authority, isi);
+
+                if verdict.is_deny() {
+                    return verdict;
+                }
+
+                isi.execute()
+            }
+
+            pass!()
+        }
+    }
+}
+
+pub fn validate_query<V: Validate + ?Sized>(
+    validator: &mut V,
+    authority: &AccountId,
+    query: &QueryBox,
+) -> Verdict {
+    use QueryBox::*;
+
+    macro_rules! query_validators {
+        ( $($validator:ident($query:ident)),+ $(,)? ) => {
+            match query { $(
+                $query(query) => $validator(validator, authority, query), )+
+            }
+        };
+    }
```

Comment:
```suggestion
    macro_rules! query_validators {
        ( $($validator:ident($query:ident)),+ $(,)? ) => {
            match query { $(
                $query(query) => validator.$validator(authority, query), )+
            }
        };
    }
```

Shouldn't it be like so? So that we use user-provided methods instead of our defaults

Reply:
Maybe same in other places like that

Reply:
I think there are errors in some places